### PR TITLE
Refactor pkg/storage

### PR DESCRIPTION
**Pull request by qbarrand**
This PR:
- refactors `pkg/storage` internals to use Kubernetes native types instead of `unstructured.Unstructured`;
- makes `pkg/storage` expose an interface;
- replace `pkg/storage` usage with the interface.

This is a follow-up to https://github.com/openshift-psap/special-resource-operator/pull/197.

/cc @pmtk @yevgeny-shnaidman

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -2,122 +2,96 @@ package storage
...
-		return err
-	}
+	err := s.kubeClient.Get(context.TODO(), dep, cm)
 
-	err = clients.Interface.Update(context.TODO(), cm)
-	if err != nil {
+	if apierrors.IsNotFound(err) {
 		warn.OnError(err)
-		return err
+		return cm, err
```
Comment (pmtk) [inline pkg/storage/storage.go]: Nit: any reason `cm` is returned instead of `nil`?

**Review by pmtk — COMMENTED** [review]
beside that comment, looks good to me

File: pkg/storage/storage.go (hunk 1 above)
Comment (qbarrand) [reply pkg/storage/storage.go]: Nice catch! While calling functions should anyway not use the return value, I changed it to `nil`.

**Pull request merged**